bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/api/RestUtils.java

```diff
@@ -0,0 +1,193 @@
+        if (params != null) {
+            params.forEach(request::param);
+        }
+        if (headers != null) {
+            headers.forEach(request::header);
+        }
+
+        if (logger.isTraceEnabled()) {
+            logger.trace("GET request: {}", request.getURI());
+        }
+        try {
+            ContentResponse response = request.send();
+
+            logger.trace("GET response: {}", response.getContentAsString());
+
+            return new RestResult(response.getContentAsString(), response.getStatus());
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.error("Exception occurred during GET execution: {}", e.getMessage(), e);
+            throw new CommunicationException(e);
+        }
+    }
+
+    @Nullable
+    public static RestResult postCall(HttpClient httpClient, String encodedUrl, Map<String, String> headers,
+            String jsonData) throws IOException {
+        try {
+            return postCall(httpClient, encodedUrl, headers, new StringContentProvider(jsonData));
+        } catch (UnsupportedEncodingException e) {
+            logger.error(
+                    "Unexpected error. Character encoding from json informed not supported by this platform. Payload:{}",
+                    jsonData, e);
+            throw new IllegalStateException(
+                    "Unexpected error. Character encoding from json informed not supported by this platform.", e);
+        }
+    }
+
+    @Nullable
+    public static RestResult postCall(HttpClient httpClient, String encodedUrl, Map<String, String> headers,
+            Map<String, String> formParams) throws IOException {
+        Fields fields = new Fields();
+        formParams.forEach(fields::put);
+        try {
+            return postCall(httpClient, encodedUrl, headers, new FormContentProvider(fields));
+        } catch (UnsupportedEncodingException e) {
+            logger.error(
+                    "Unexpected error. Character encoding received from Form Parameters not supported by this platform. Form Parameters:{}",
+                    formParams, e);
+            throw new IllegalStateException(
+                    "Unexpected error. Character encoding received from Form Parameters not supported by this platform.",
+                    e);
+        }
+    }
+
+    @Nullable
+    private static RestResult postCall(HttpClient httpClient, String encodedUrl, Map<String, String> headers,
+            ContentProvider contentProvider) throws IOException {
+
+        try {
+            Request request = httpClient.newRequest(encodedUrl).method("POST").content(contentProvider).timeout(10,
+                    TimeUnit.SECONDS);
+            if (headers != null) {
+                headers.forEach(request::header);
+            }
+            if (logger.isTraceEnabled()) {
+                logger.trace("POST request: {}", request.getURI());
+            }
```

Comment:
debug can be used for uri metadata. Only when logging verbose content request/resposne you should lower to trace.
```suggestion
            logger.debug("POST request: {}", request.getURI());
```